src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+package mate.academy.dao;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES (?, ?)";
+
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(sql,
+                Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setInt(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Creating book failed, no rows affected.");
```

Comment:
Let's use more informative messages for exceptions. Instead of `Creating book failed, no rows affected.`, we can use `Creating book failed, no rows affected for book: ` + book